what4/src/What4/Expr/Builder.hs

```diff
@@ -868,44 +867,44 @@ instance Pretty (Expr t tp) where
 
 
 -- | @AppPPExpr@ represents a an application, and it may be let bound.
-data AppPPExpr
+data AppPPExpr ann
    = APE { apeIndex :: !PPIndex
          , apeLoc :: !ProgramLoc
          , apeName :: !Text
-         , apeExprs :: ![PPExpr]
+         , apeExprs :: ![PPExpr ann]
```

Comment:
This seems like a good place to add a definite expression-term annotation rather than using an parameterized version which doesn't allow any of the what4 expression information to be provided.

Reply:
None of the what4 code uses `Doc` annotations yet; my thinking is that as long as that's the case, keeping pretty-printing code polymorphic in the annotation type is preferable to fixing the type to `Void`. That way, if someone wants to start using annotations in some part of what4 in the future, the polymorphic `Doc` code will continue to work with it unmodified.

In any case, I'm not going to be adding any new `Doc` annotations in this PR :)

Reply:
My thinking was that instead of `Void` annotations that at some point we'd have a definite What4 annotation type like `data What4Ann = What4_FreshVar | What4_BVOperator | ...` and that the pretty printer instances here would supply those annotations as they pprinted the Expr.   That said, limiting the scope of this PR is definitely reasonable.

Reply:
That sounds like a good idea for a future PR. Do you want to record those thoughts on a what4 ticket?

Reply:
Done.